src/Splitter.sol

```diff
@@ -0,0 +1,113 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface VatLike {
+    function move(address, address, uint256) external;
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface DaiJoinLike {
+    function vat() external view returns (address);
+    function exit(address, uint256) external;
+}
+
+interface FlapLike {
+    function kick(uint256, uint256) external returns (uint256);
+    function cage(uint256) external;
+}
+
+interface FarmLike {
+    function notifyRewardAmount(uint256 reward) external;
+}
+
+contract Splitter {
+    mapping (address => uint256) public wards;
+
+    VatLike     public immutable vat;
+    DaiJoinLike public immutable daiJoin;
+    FarmLike    public immutable farm;
+
+    FlapLike public flapper;
+    uint256  public burn; // in WAD; 1 WAD = 100% burn
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(
+        address _daiJoin,
+        address _farm
+    ) {
+        daiJoin = DaiJoinLike(_daiJoin);
+        vat     = VatLike(daiJoin.vat());
+        farm    = FarmLike(_farm);
+        
+        vat.hope(_daiJoin);
+        
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "Splitter/not-authorized");
+        _;
+    }
+
+    uint256 internal constant WAD = 10 ** 18;
+    uint256 internal constant RAY = 10 ** 27;
+
+    function rely(address usr) external auth { wards[usr] = 1; emit Rely(usr); }
+    function deny(address usr) external auth { wards[usr] = 0; emit Deny(usr); }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if   (what == "burn") burn = data;
+        else revert("Splitter/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "flapper") {
+            vat.nope(address(flapper));
+            flapper = FlapLike(data);
+            vat.hope(data);
+        }
+        else revert("Splitter/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    // Note: the kick() cadence is determined by flapper.hop()
+    function kick(uint256 tot, uint256) external auth returns (uint256) {
+        vat.move(msg.sender, address(this), tot);
+
+        uint256 lot = tot * burn / WAD;
+        flapper.kick(lot, 0);
+
+        FarmLike farm_ = farm;
+        uint256 pay = (tot - lot) / RAY;
```

Comment:
Not sure if we should call the farming logic if `pay == 0`.
Need to test it but I think it would just extend to a new duration with nothing new to pay.

Reply:
Fixed